### ISSUE TITLE
GPUParticleSystem.js color option bug and memory leak

### ISSUE THREAD

**Issue by vorkulsky** [mrdoob/three.js#11149]
There are two problems in GPUParticleSystem.js:

1) spawnParticle method color option is not work as a valid rgb color. There is an error in sending a color to the shader.
2) The dispose method is missing. Therefore, a GPUParticleSystem memory is not cleared.
The dispose method should contain:
```
self.particleShaderMat.dispose();
self.particleShaderGeo.dispose();
```

**Comment by Mugen87** [comment]
> 1. spawnParticle method color option is not work as a valid rgb color. There is an error in sending a color to the shader.

Right now the color needs to be defined as hexadecimal numbers like this. How do you pass the color value so it's producing an error?
```javascript
options = { color: 0xff0000 }
```

**Comment by vorkulsky** [comment]
In example https://threejs.org/examples/#webgl_gpu_particle_system color option is equal 0xaa88ff.
I see green color, but rgb #aa88ff shoud is purple.

**Comment by Mugen87** [comment]
Oh, that's strange indeed! I'm getting this:

![image](https://cloud.githubusercontent.com/assets/12612165/24837074/4ea2fa96-1d2c-11e7-8a22-b50cd5c539ad.png)

> There is an error in sending a color to the shader.

Can you post this error message please?

**Comment by jostschmithals** [comment]
I see the correct color, too (Chrome, FF, Edge; Windows 10)

**Comment by Mugen87** [comment]
@vorkulsky I've added a `dispose` method that regards the material, geometries and textures. The mentioned color problem is still an open issue. I assume there is a problem wit the float <-> vector4 encoding/decoding on your machine.

**Comment by vorkulsky** [comment]
![1](https://cloud.githubusercontent.com/assets/2943099/24837440/ae48ee1c-1d4d-11e7-8017-a5c01c1db2b5.png)
Chrome, FF, Edge; Windows 10, Intel hd graphics 4000

**Comment by jostschmithals** [comment]
This seems to be a similar issue: https://github.com/mrdoob/three.js/issues/7804

**Comment by Mugen87** [comment]
One option to solve this problem: We avoid the whole float/vector4 encoding stuff and use a single shader attribute for each property. This would also make the code more simple.

**Comment by jostschmithals** [comment]
I just saw: On my (very low end) laptop (intel hd) the example is displayed in a nice green color, too.

**Comment by Mugen87** [comment]
@jostschmithals @vorkulsky Can you please test if  this version of `GPUParticleSystem` renders the colors are correctly on your devices?

[GPUParticleSystem.js.zip](https://github.com/mrdoob/three.js/files/908269/GPUParticleSystem.js.zip)

**Comment by jostschmithals** [comment]
@Mugen87 Using your test code the color is purple on the prehistoric device that displayed green before.

**Comment by Mugen87** [comment]
Good to hear that! It actually looks like that the float/vector4 encoding is problematic on older devices. As mentioned in this [comment](https://github.com/mrdoob/three.js/issues/7804#issuecomment-172097179) the floating point precision probably causes this issue.

**Comment by Mugen87** [comment]
Fixed with #11151.
